modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/ConfigurationInstanceProviderFactoryBean.java

```diff
@@ -89,4 +85,30 @@ public boolean isSingleton()
     {
         return true;
     }
+
+    private DynamicConfigPolicy getDynamicConfigPolicy(ElementDescriptor element)
+    {
+        ElementDescriptor policyElement = element.getChildByName("dynamic-config-policy");
+        if (policyElement == null)
+        {
+            return NullDynamicConfigPolicy.INSTANCE;
+        }
+
+        return new ImmutableDynamicConfigPolicy(getExpirationPolicy(policyElement));
+    }
+
+    private ExpirationPolicy getExpirationPolicy(ElementDescriptor dynamicConfigPolicyElement)
+    {
+        ElementDescriptor expirationPolicyElement = dynamicConfigPolicyElement.getChildByName("expiration-policy");
+        if (expirationPolicyElement == null)
+        {
+            return new ImmutableExpirationPolicy(5, TimeUnit.MINUTES, TimeSupplier.INSTANCE);
```

Comment:
Being able to use a testTimeSupplier is not useful only for unit tests. During a functional/integration test using this supplier could make more easy to write tests that could require complex synchronization using latches otherwise